verify.go

```diff
@@ -6,22 +6,11 @@ import (
 	"fmt"
 
 	intoto "github.com/in-toto/in-toto-golang/in_toto"
-	"github.com/openpubkey/openpubkey/parties"
+	"github.com/openpubkey/openpubkey/client"
 	"github.com/secure-systems-lab/go-securesystemslib/dsse"
 )
 
-func VerifyInTotoEnvelope(ctx context.Context, env dsse.Envelope, oidcProvider OIDCProvider) (*intoto.Statement, error) {
-	var provider parties.OpenIdProvider
-	switch oidcProvider {
-	case GithubActionsOIDC:
-		var err error
-		provider = parties.NewGithubOp("", "")
-		if err != nil {
-			return nil, err
-		}
-	default:
-		return nil, fmt.Errorf("unkown oidc provider %v", oidcProvider)
-	}
+func VerifyInTotoEnvelope(ctx context.Context, env dsse.Envelope, provider client.OpenIdProvider) (*intoto.Statement, error) {
```

Comment:
While we're changing this interface, we could change the envelope to a pointer type to match the return value of `SignInTotoStatement`:
```suggestion
func VerifyInTotoEnvelope(ctx context.Context, env *dsse.Envelope, provider client.OpenIdProvider) (*intoto.Statement, error) {
```